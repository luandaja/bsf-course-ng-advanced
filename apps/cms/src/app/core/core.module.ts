import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { StoreModule, ActionReducerMap } from '@ngrx/store';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { environment } from '../../environments/environment';
import { StoreRouterConnectingModule, RouterState } from '@ngrx/router-store';
import { CustomSerializer } from '../store/router';

@NgModule({
  declarations: [],
  imports: [
    CommonModule,
    StoreModule.forRoot(
      {},
      {
        runtimeChecks: {
          strictActionImmutability: true,
          strictActionSerializability: true,
          strictStateImmutability: true,
          strictStateSerializability: true
        }
      }
    ),
    StoreDevtoolsModule.instrument({ maxAge: 25, logOnly: environment.production })
  ]
})
export class CoreModule {}
