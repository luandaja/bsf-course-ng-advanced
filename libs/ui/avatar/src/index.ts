export * from './lib/ui-avatar.module';
export * from './models';

