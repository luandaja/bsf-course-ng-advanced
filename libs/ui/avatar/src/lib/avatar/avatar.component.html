<img
	[src]="source"
	[width]="width"
	[height]="height"
	[ngClass]="{ selected: isSelected }"
	class="rounded-circle border-grey img-fluid pb-2"
/>
