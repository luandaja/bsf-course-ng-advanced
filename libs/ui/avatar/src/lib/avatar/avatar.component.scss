.selected {
  opacity: 0.4;
}

img:hover {
  transform: scale(1.04);
  box-shadow: 0 3px 12px rgba(0, 0, 0, 0.23), 0 3px 12px rgba(0, 0, 0, 0.16);
}
