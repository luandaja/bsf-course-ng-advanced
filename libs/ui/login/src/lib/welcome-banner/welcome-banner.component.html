<section class="welcome-container">
	<div class="d-flex flex-column m-3 h-100">
		<div class="p-2 mb-auto">
			<span class="brand-name">{{ banner.upperText }}</span>
		</div>
		<div class="p-2 align-self-center" *ngIf="banner.imageUrl">
			<gt-avatar [width]="500" [source]="banner.imageUrl"></gt-avatar>
		</div>
		<div class="message mt-auto">
			<h2>{{ banner.title }}</h2>
			<span>{{ banner.contentText }}</span>
		</div>
	</div>
</section>
