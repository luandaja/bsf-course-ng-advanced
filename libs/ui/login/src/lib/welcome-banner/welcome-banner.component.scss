.welcome-container {
  background-color: #f89937;
  color: white;
  min-height: 211px;
  clip-path: polygon(75% 0%, 100% 50%, 75% 100%, 0% 100%, 0 49%, 0% 0%);
  padding: 20px;
  height: 100vh;
  padding-right: 25%;
  background-image: url("https://www.tercetocomunicacion.es/wp-content/uploads/2017/06/portada.jpg");
  background-size: 93% 101%;
}

.brand-name {
  font-weight: 400;
  font-style: italic;
}
.message {
  background-color: #212f3d;
  opacity: 0.8;
  border-radius: 19px;
  width: w-50;
  padding: 3rem;
}
@media only screen and (max-width: 600px) {
  .welcome-container {
    height: 40vh;
    margin-bottom: 15px;
    background-size: 91% 72%;
  }
  .message {
    padding: 0.25rem;
    width: w-100;
  }
}
