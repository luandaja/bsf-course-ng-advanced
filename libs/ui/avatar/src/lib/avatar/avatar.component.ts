import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';

@Component({
	selector: 'gt-avatar',
	templateUrl: './avatar.component.html',
	styleUrls: ['./avatar.component.scss']
})
export class AvatarComponent implements OnInit {
	protected defaultImg = 'https://m.media-amazon.com/images/S/aplus-media/mg/dbf4301f-af40-46f2-9a87-a99deddcd9a2._SL300__.jpg';
	protected _source = this.defaultImg;

	@Input() isSelected = false;
	@Input() width = 30;
	@Input() height = 30;
	@Input()
	get source() {
		return this._source;
	}

	set source(value) {
		this._source = value || this.defaultImg;
	}

	constructor() { }

	ngOnInit() { }

}
