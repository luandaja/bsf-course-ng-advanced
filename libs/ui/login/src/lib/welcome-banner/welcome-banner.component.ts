import { Component, Input } from '@angular/core';
import { Banner } from '../models/Banner';

@Component({
	selector: 'gt-welcome-banner',
	templateUrl: './welcome-banner.component.html',
	styleUrls: ['./welcome-banner.component.scss']
})
export class WelcomeBannerComponent {
	@Input() banner: Banner;

	constructor() {}
}
