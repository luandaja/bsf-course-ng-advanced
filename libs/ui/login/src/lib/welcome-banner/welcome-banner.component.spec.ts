import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { WelcomeBannerComponent } from './welcome-banner.component';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { UiAvatarModule } from '@glotrix/ui/avatar';

describe('WelcomeBannerComponent', () => {
	let component: WelcomeBannerComponent;
	let fixture: ComponentFixture<WelcomeBannerComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations: [WelcomeBannerComponent],
			schemas: [CUSTOM_ELEMENTS_SCHEMA]
		})
			.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(WelcomeBannerComponent);
		component = fixture.debugElement.componentInstance;

	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('should match snapshot', () => {

		component.banner = {
			contentText: 'content', imageUrl: 'url',
			title: 'my title', upperText: 'my brand name'
		};
		fixture.detectChanges();

		expect(fixture).toMatchSnapshot();
	});

});
