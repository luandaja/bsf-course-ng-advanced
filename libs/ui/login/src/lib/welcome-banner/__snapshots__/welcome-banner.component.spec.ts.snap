// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeBannerComponent should match snapshot 1`] = `
<gt-welcome-banner
  banner={[Function Object]}
>
  <section
    class="welcome-container"
  >
    <div
      class="d-flex flex-column m-3 h-100"
    >
      <div
        class="p-2 mb-auto"
      >
        <span
          class="brand-name"
        >
          my brand name
        </span>
      </div>
      <div
        class="p-2 align-self-center"
      >
        <gt-avatar />
      </div>
      <div
        class="p-2 mt-auto w-50"
      >
        <h2>
          my title2
        </h2>
        <span>
          content
        </span>
      </div>
    </div>
  </section>
</gt-welcome-banner>
`;

exports[`WelcomeBannerComponent should match snapshot2 1`] = `
<gt-welcome-banner
  banner={[Function Object]}
>
  <section
    class="welcome-container"
  >
    <div
      class="d-flex flex-column m-3 h-100"
    >
      <div
        class="p-2 mb-auto"
      >
        <span
          class="brand-name"
        >
          my brand name2
        </span>
      </div>
      <div
        class="p-2 align-self-center"
      >
        <gt-avatar />
      </div>
      <div
        class="p-2 mt-auto w-50"
      >
        <h2>
          my title22
        </h2>
        <span>
          content2
        </span>
      </div>
    </div>
  </section>
</gt-welcome-banner>
`;
